src/derivative_utils.jl

```diff
@@ -237,24 +251,23 @@ function LinearAlgebra.mul!(Y::AbstractVecOrMat, W::WOperator, B::AbstractVecOrM
   end
 end
 
-function do_newJ(integrator, alg::T, cache, repeat_step)::Bool where T
+function do_newJ(integrator, alg::T, cache, repeat_step)::Bool where T # any changes here need to be reflected in FIRK
   repeat_step && return false
   !alg_can_repeat_jac(alg) && return true
   isnewton = T <: NewtonAlgorithm
-  isnewton && (T <: RadauIIA5 ? ( @unpack ηold,nl_iters = cache ) : ( @unpack ηold,nl_iters = cache.nlsolver ))
-  integrator.force_stepfail && return true
+  isnewton && (T <: RadauIIA5 ? ( nlstatus = cache.status ) : ( nlstatus = get_status(cache.nlsolver) ))
+  nlsolvefail(nlstatus) && return true
   # reuse J when there is fast convergence
-  fastconvergence = nl_iters == 1 && ηold <= alg.new_jac_conv_bound
+  fastconvergence = nlstatus === FastConvergence
```

Comment:
why ===?

Reply:
I think that `===` is a more performant way of doing the comparison if applicable.